static/app/routes.spec.tsx

```diff
@@ -0,0 +1,119 @@
+import {createRoutes, RouteComponent} from 'react-router';
+
+import * as constants from 'sentry/constants';
+import {buildRoutes} from 'sentry/routes';
+
+import {normalizeUrl} from './utils/withDomainRequired';
+
+// Setup a module mock so that we can replace
+// usingCustomerDomain with a getter.
+jest.mock('sentry/constants', () => {
+  const originalModule = jest.requireActual('sentry/constants');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    get usingCustomerDomain() {
+      return false;
+    },
+  };
+});
+
+// Workaround react-router PlainRoute type not covering redirect routes.
+type RouteShape = {
+  childRoutes?: RouteShape[];
+  component?: RouteComponent;
+  from?: string;
+  path?: string;
+};
+
+type RouteMetadata = {
+  leadingPath: string;
+  route: RouteShape;
+};
+
+function extractRoutes(rootRoute: any): Record<string, RouteComponent> {
+  const routeTree = createRoutes(rootRoute);
+  const routeMap: Record<string, RouteComponent> = {};
+
+  // A queue of routes we need to visit
+  const visitQueue: RouteMetadata[] = [{leadingPath: '', route: routeTree[0]}];
+  while (visitQueue.length > 0) {
+    const current = visitQueue.pop();
+    if (!current) {
```

Comment:
Was this to overcome typescript's limitations? We already know that `visitQueue.length > 0` by line 43.

Reply:
Yes typescript was complaining about the possibility of undefined.